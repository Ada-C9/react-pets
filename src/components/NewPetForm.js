import React, { Component } from 'react';
import PropTypes from 'prop-types';

class NewPetForm extends Component {
  static propTypes = {
    addPetCallback: PropTypes.func.isRequired
  };

  constructor() {
    super();

    this.state = {
      name: '',
      breed: '',
      age: '',
      about: ''
    };
  }

  onInputChange = (event) => {
    let updatedInput = {};
    updatedInput[event.target.name] = event.target.value;
    this.setState(updatedInput);
    // "name" : event.target.value
    // event.target.name
    // event.target.value
  }

  onFormSubmit = (event) => {
    event.preventDefault();
    console.log(this.state);
    this.props.addPetCallback(this.state);

    this.setState({
      name: '',
      breed: '',
      age: '',
      about: ''
    });
  }

  render() {
    return (
      <form onSubmit={this.onFormSubmit}>
        <h2>Test header</h2>
        <div>
          <label htmlFor="name">Name</label>
          <input type="text"
            name="name"
            value={this.state.name}
            onChange={this.onInputChange}/>
        </div>
        <div>
          <label htmlFor="breed">Breed</label>
          <input type="text"
            name="breed"
            value={this.state.breed}
            onChange={this.onInputChange}/>
        </div>
        <div>
          <label htmlFor="age">Age</label>
          <input type="text"
            name="age"
            value={this.state.age}
            onChange={this.onInputChange}/>
        </div>
        <div>
          <label htmlFor="about">About</label>
          <input type="text"
            name="about"
            value={this.state.about}
            onChange={this.onInputChange}/>
        </div>
        <div><input type="submit"/>
        </div>
      </form>
    );
  }
}

export default NewPetForm;
