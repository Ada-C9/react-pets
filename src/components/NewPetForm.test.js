import React from 'react';
import { mount, shallow } from 'enzyme';

import NewPetForm from './NewPetForm';

describe('NewPetForm', () => {
  test('that it matches an existing snapshot', () => {
    // Arrange
    // Mount the component in the DOM
    // Deep Rendering
    const wrapper = shallow(<NewPetForm
      addPetCallback={() => {} }
      />);

    // Assert
    expect(wrapper).toMatchSnapshot();
    // Remove the component from the DOM
    // Clean up Clean up Everybody..
    wrapper.unmount();
  });

  test('When a user enters a name in a \
  text field the field is updated', () => {
    // Arrange
    // Shallow Mounted the wrapper
    const wrapper = shallow(<NewPetForm
      addPetCallback={() => {} }
      />);

    // find the input field
    let nameField =
        wrapper.find('input[name="name"]');

    // Act
    nameField.simulate('change', {
      target: {
        name: 'name',
        value: 'Bob',
      },
    });
    // Force the onChange event
    wrapper.update();
    nameField =
        wrapper.find('input[name="name"]');

    // Assert
    expect(nameField.getElement().props.value)
      .toEqual('Bob');

  });

  test('when the user types on a field the value changes', () => {
    // Arrange
    // Shallow Mounted the wrapper
    const wrapper = shallow(<NewPetForm
      addPetCallback={() => {} }
      />);

    const fields = [
      {
        field: 'name',
        value: 'Bob',
      },
      {
        field: 'age',
        value: 3,
      },
      {
        field: 'breed',
        value: 'mutt',
      },
      {
        field: 'about',
        value: 'great mutt!',
      }
    ];


    fields.forEach(({field, value}) => {
      // find the input field
      let nameField =
          wrapper.find(`[name="${field}"]`);

      // Act
      nameField.simulate('change', {
        target: {
          name: field,
          value,
        },
      });
      // Force the onChange event
      wrapper.update();
      nameField =
          wrapper.find(`[name="${field}"]`);

      // Assert
      expect(nameField.getElement().props.value)
        .toEqual(value);

    });
  });









});
