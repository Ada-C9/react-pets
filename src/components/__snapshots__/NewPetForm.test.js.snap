// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPetForm that it matches an existing snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <div>
    <label
      htmlFor="name"
    >
      Name: 
    </label>
    <input
      id="name"
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div>
    <label
      htmlFor="age"
    >
      Age: 
    </label>
    <input
      id="age"
      name="age"
      onChange={[Function]}
      type="number"
      value={0}
    />
  </div>
  <div>
    <label
      htmlFor="breed"
    >
      Breed: 
    </label>
    <input
      id="breed"
      name="breed"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div>
    <label
      htmlFor="about"
    >
      About: 
    </label>
    <textarea
      id="about"
      name="about"
      onChange={[Function]}
      value=""
    />
  </div>
  <input
    type="submit"
    value="Add Pet"
  />
</form>
`;
